aten/src/ATen/core/op_registration/op_registration_test.cpp

```diff
@@ -219,4 +220,265 @@ TEST(OperatorRegistrationTest, givenOpWithMultipleKernels_whenKernelsHaveSameDis
   }, "Tried to register multiple kernels with same dispatch key");
 }
 
+
+/**
+ * This is used to check that a given type works correctly when passed as input
+ * to or as output from a kernel.
+ *
+ * Call ArgTypeTestKernel<Input, Output>::test(input, inputExpectation, output, outputExpectation, schema)
+ * to test that a kernel with `Input` as input type and `Output` as output types,
+ * when called with `input` fulfills `inputExpectation` inside the kernel, then
+ * returns `output` and the returned value fulfills `outputExpectation`.
```

Comment:
"fulfills `inputExpectation`" is not very clear and I had to look at the signatures to figure out what you meant by this.

Reply:
I added a paragraph to explain it a bit better